packages/scene-composer/src/store/helpers/serializationHelpers.ts

```diff
@@ -573,6 +585,53 @@ function createDocumentState(
     );
   }
 
+  if (scene.properties) {
+    if (scene.properties[KnownSceneProperty.FogCustomColors]) {
+      scene.properties[KnownSceneProperty.FogCustomColors] = (
+        scene.properties[KnownSceneProperty.FogCustomColors] as string[]
+      ).filter((color) => isValidHexCode(color));
+    }
+    if (scene.properties[KnownSceneProperty.BackgroundCustomColors]) {
+      scene.properties[KnownSceneProperty.BackgroundCustomColors] = (
+        scene.properties[KnownSceneProperty.BackgroundCustomColors] as string[]
+      ).filter((color) => isValidHexCode(color));
+    }
+    if (scene.properties[KnownSceneProperty.GroundCustomColors]) {
+      scene.properties[KnownSceneProperty.GroundCustomColors] = (
+        scene.properties[KnownSceneProperty.GroundCustomColors] as string[]
+      ).filter((color) => isValidHexCode(color));
+    }
```

Comment:
These all do the same thing, and you could simplify this code. You can likely remove a lot of conditionals if you instead setup a default state.

For instance, instead of checking `if scene.properties`, initialize it.

It's odd that this logic is mutating the original scene object, it would be cleaner if the initial shape of the scene.properties was a known state/structure, and that the mutation happens in the store, instead of correcting it on save. But even barring that, if we need to map it to a different structure for writing, it should be copied instead of mutated:

```
const persistedscene = Object.keys(scene).reduce((persist, property) => {
      persist[property] = property || [] as String[]; // basic example to show
}, {});

return JSON.serialize(persistedscene);
```